src/view/LoggedInView.java

```diff
@@ -0,0 +1,152 @@
+    public LoggedInView(LoggedInViewModel loggedInViewModel, LogoutViewModel logoutViewModel, LogoutController logoutController) {
+        this.loggedInViewModel = loggedInViewModel;
+        this.loggedInViewModel.addPropertyChangeListener(this);
+        this.logoutViewModel = logoutViewModel;
+        this.logoutController = logoutController;
+
+        logoutViewModel.addPropertyChangeListener(this);
+
+        JLabel title = new JLabel("Profile");
+        title.setAlignmentX(Component.CENTER_ALIGNMENT);
+
+        JLabel usernameInfo = new JLabel("Currently logged in: ");
+        username = new JLabel();
+
+        JPanel buttons = new JPanel();
+        logout = new JButton(loggedInViewModel.LOGOUT_BUTTON_LABEL);
+        buttons.add(logout);
+
+        requests = new JButton(loggedInViewModel.REQUESTS_BUTTON_LABEL);
+        buttons.add(requests);
+
+        friends = new JButton(loggedInViewModel.FRIENDS_BUTTON_LABEL);
+        buttons.add(friends);
+
+        match = new JButton(loggedInViewModel.MATCH_BUTTON_LABEL);
+        buttons.add(match);
+
+        editProfile = new JButton(loggedInViewModel.EDIT_PROFILE_BUTTON_LABEL);
+        buttons.add(editProfile);
+
+        logout.addActionListener(this);
+        logout.addActionListener(
+            // This creates an anonymous subclass of ActionListener and instantiates it.
+            new ActionListener() {
+                public void actionPerformed(ActionEvent evt) {
+                    if (evt.getSource().equals(logout)) {
+                        logoutController.execute();
+                    }
+                }
+            }
+        );
+
+
+        requests.addActionListener(this);
+        requests.addActionListener(
+            // This creates an anonymous subclass of ActionListener and instantiates it.
+            new ActionListener() {
+                public void actionPerformed(ActionEvent evt) {
+                    if (evt.getSource().equals(requests)) {
+                        displayRequestsController.execute();
+                    }
+                }
+            }
+        );
+
+        friends.addActionListener(this);
+        friends.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                new ActionListener() {
+                    public void actionPerformed(ActionEvent evt) {
+                        if (evt.getSource().equals(friends)) {
+                            displayFriendsController.execute();
+                        }
+                    }
+                }
+        );
+
+        match.addActionListener(this);
+        match.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                new ActionListener() {
+                    public void actionPerformed(ActionEvent evt) {
+                        if (evt.getSource().equals(match)) {
+                            matchController.execute();
```

Comment:
matchController.execute() method should take in a User Object as it is required in the matching use case. To obtain the current User object, we can potentially create a variable, currentState, and set it equal to LoggedInViewModel.getState(), then use a getUser() method from the currentState.